src/registry/map/preview.tsx

```diff
@@ -37,10 +39,27 @@ const Preview: React.FC<ComponentPreviewProps<MapComponentSchema>> = ({component
     validate = {},
     defaultZoom,
     initialCenter = {},
+    tileLayerIdentifier,
   } = component;
+  const {getMapTileLayers} = useContext(BuilderContext);
+  const {value: options, loading, error} = useAsync(async () => await getMapTileLayers(), []);
+  if (error) {
+    throw error;
+  }
+
   const {required = false} = validate;
   const {lat = 52.1326332, lng = 5.291266} = initialCenter;
   const zoom = defaultZoom ?? 8;
+
+  const tileLayerUrl = (): string => {
+    if (!tileLayerIdentifier || loading || !options?.length) {
+      return TILE_LAYER_RD.url;
+    }
```

Comment:
I think it's a bit odd the tile layer can change because we're still loading them. Perhaps it's nicer to just display a loader rather than any map at all while we're still fetching the available tile layers? (referring to the `loading` var)

Reply:
Check, yeah good point.

I've added a simple Loader component, using the basic bootstrap spinner. This will be shown, when the data is still loading.